src/PockyBot.NET/Services/Pegs/PegResultsHelper.cs

```diff
@@ -0,0 +1,145 @@
+                    Pegs = validPegs,
+                    Penalties = penaltyPegs
+                };
+            }).Where(x => x.PegCount > 0 || x.PenaltyCount > 0)
+                .ToList();
+        }
+
+        public List<PegRecipient> GetWinners(List<PegRecipient> allRecipients)
+        {
+            var minimum = _configRepository.GetGeneralConfig("minimum") ?? 0;
+            var winners = _configRepository.GetGeneralConfig("winners") ?? 3;
+            var eligibleWinners = allRecipients.Where(x => x.PegsGivenCount >= minimum).ToList();
+            var topCutoff = eligibleWinners.Select(x => x.TotalPoints)
+                .OrderByDescending(x => x)
+                .Take(winners)
+                .LastOrDefault();
+
+            return eligibleWinners.Where(x => x.TotalPoints >= topCutoff)
+                .OrderBy(x => x.TotalPoints)
+                .ToList();
+        }
+
+        public List<PegCategory> GetCategories(List<PegRecipient> allRecipients)
+        {
+            var keywords = _configRepository.GetStringConfig("keyword").ToArray();
+            return keywords.Select(x =>
+            {
+                var categoryRecipients = MapRecipientsToCategoryRecipients(allRecipients, x);
+
+                if (categoryRecipients.Count == 0)
+                {
+                    return new PegCategory
+                    {
+                        Name = x,
+                        Recipients = new List<PegRecipient>()
+                    };
+                }
+
+                var topRecipients = categoryRecipients.Where(y => y.TotalPoints == categoryRecipients[0].TotalPoints)
+                    .ToList();
+
+                return new PegCategory
+                {
+                    Name = x,
+                    Recipients = topRecipients
+                };
+            }).ToList();
+        }
+
+        private List<PegDetails> GetValidPegs(List<Peg> pegs, int? requireKeywords, string[] keywords,
+            string[] penaltyKeywords, string receiverLocation)
+        {
+            return pegs.Where(y => _pegHelper.IsPegValid(y.Comment, requireKeywords, keywords, penaltyKeywords))
+                .Select(y => new PegDetails
+                {
+                    SenderName = y.Sender.Username,
+                    Weight = _pegHelper.GetPegWeighting(y.Sender.Location?.Location, receiverLocation),
+                    Comment = y.Comment,
+                    Keywords = keywords.Where(z => y.Comment.Contains(z)).ToList(),
+                    SenderLocation = y.Sender.Location?.Location
+                })
+                .ToList();
+        }
+
+        private List<PegDetails> GetPenaltyPegs(List<Peg> pegs, int? requireKeywords, string[] keywords,
+            string[] penaltyKeywords)
+        {
+            return pegs.Where(y => !_pegHelper.IsPegValid(y.Comment, requireKeywords, keywords, penaltyKeywords))
+                .Select(y => new PegDetails
```

Comment:
Personally prefer using inheritance/interfaces to separate the different type of pegs.

Not sure where these functions are used but it might make sense combining these 2 methods and push the IsPegValid check further down.

Maybe move the mapper into a helper class or a ToPegDetail method inside peg